packages/flutter/lib/src/widgets/viewport.dart

```diff
@@ -61,8 +61,37 @@ class Viewport extends MultiChildRenderObjectWidget {
   }) : assert(offset != null),
        assert(slivers != null),
        assert(center == null || slivers.where((Widget child) => child.key == center).length == 1),
+       _autoCache = false,
        super(key: key, children: slivers);
 
+  /// Creates a widget that is bigger on the inside.
+  ///
+  /// The viewport listens to the [offset], which means you do not need to
+  /// rebuild this widget when the [offset] changes.
+  ///
+  /// The [offset] argument must not be null.
+  ///
+  /// A cache extent for this viewport will be automatically calculated during
+  /// layout as triple the main axis extent.
+  Viewport.autoCaching({
```

Comment:
Why do we need a separate constructor for this instead of just making this accessible through the main constructor?